python/ray/autoscaler/updater.py

```diff
@@ -255,14 +257,25 @@ def run(self,
         if cmd:
             logger.info(self.log_prefix +
                         "Running {} on {}...".format(cmd, self.ssh_ip))
-            final_cmd += with_interactive(cmd)
+            final_cmd += with_interactive(cmd) if interactive else [quote(cmd)]
         else:
             # We do this because `-o ControlMaster` causes the `-N` flag to
             # still create an interactive shell in some ssh versions.
-            final_cmd.append("while true; do sleep 86400; done")
+            final_cmd.append(quote("while true; do sleep 86400; done"))
+
         try:
             if with_output:
-                return self.process_runner.check_output(final_cmd)
+                if interactive:
+                    ssh_process = self.process_runner.Popen(
+                        final_cmd, stderr=subprocess.PIPE)
+                    err = self.process_runner.check_output(
+                        ["grep", "-v", "Shared Connection to .* closed"],
+                        stdin=ssh_process.stderr)
+                    if err:
+                        print(err, file=sys.stderr)
```

Comment:
This is super hacky. Do we really need this?

Reply:
Filtering out certain error messages feels inherently hacky. 

I don't have a strong opinion about this, so I'll let you you and @richardliaw decide?

Reply:
If there's no cleaner solution, let's remove it